modules/react-native-status/ios/RCTStatus/Utils.m

```diff
@@ -110,9 +131,18 @@ + (void) migrateKeystore:(NSString *)accountData
 RCT_EXPORT_METHOD(validateMnemonic:(NSString *)seed
     callback:(RCTResponseSenderBlock)callback) {
     #if DEBUG
-        NSLog(@"validateMnemonic() method called");
+        NSLog(@"validateMnemonicV2() method called");
     #endif
-    NSString *result = StatusgoValidateMnemonic(seed);
+    NSDictionary *params = @{@"mnemonic": seed};
+    NSError *error;
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:params options:0 error:&error];
+    if (error) {
+        NSLog(@"Error creating JSON: %@", error);
+        callback(@[@"Error creating JSON"]);
```

Comment:
I'm not sure if we need to invoke the callback with a string. The native module promise expects a map in return to check if there is any error.

https://github.com/status-im/status-mobile/blob/020d5cc1c124f9bd2e8931d5275102a7b093d573/src/native_module/utils.cljs#L10

cc @clauxx